mlrun/api/db/sqldb/db.py

```diff
@@ -3030,3 +3030,142 @@ def create_data_version(self, session, version):
         now = datetime.now(timezone.utc)
         data_version_record = DataVersion(version=version, created=now)
         self._upsert(session, [data_version_record])
+
+    @retry_on_conflict
+    def store_background_task(
+        self,
+        session,
+        name: str,
+        state: str = mlrun.api.schemas.BackgroundTaskState.running,
+        project: str = None,
+        timeout: int = None,
+    ):
+        background_task_record = (
+            self._query(
+                session,
+                BackgroundTask,
+                name=name,
+                project=project,
+            )
+            .filter(BackgroundTask.project == project)
```

Comment:
Why filtering twice on the project ?

Reply:
When project is None the query ignores that, I added the `filter` to handle cases when project is None and I don't want to catch background task with the same name with project name.

Reply:
Not sure I understand, let's talk about it offline